### DISCUSSION TITLE
Is it possible to open a sound file?

### DISCUSSION THREAD

**Discussion by LanguageDescriptions** [community/community#22174] [Other Feature Feedback, Questions, & Ideas]
<p>First, some context: I am a complete newb w.r.t. Git-anything, and not a programmer. I am trying to port some plain vanilla HTML pages (from a dead server), which contain stuff like this:</p>
<p>     You can hear this in the word &lt;a href=“001.mp3”&gt;elephant&lt;/a&gt;</p>
<p>and when the reader clicks “elephant”, it plays the mp3 file (or at least does what the browser is set to do with an mp3 file). I can’t figure out if there is any translation into Markdown that does this. The following does not work as desired:</p>
<p>You can hear this in the word <a>elephant</a></p>
<p>This sends the end user to a separate page where they can click a download button or view raw, which will then play the file. Is there a way to cut out some steps, such as directly open the player without the intermediate page; or even, play it in the background or a second tab, while still displaying the main page? (End users will have minimal experience with computers, so simplicity is essential).</p>
<p>Thanks, Dave Odden</p>

**Comment by lee-dohm** [comment]
<p>Since the MP3 files are in your repository already, you don’t have to link to them via the <a href="https://github.com" rel="nofollow noopener">https://github.com</a> link. You can link to them directly via your Pages site like this:</p>
<pre><code>[elephant](/001.mp3)
</code></pre>
<p>That <code>/001.mp3</code> will be resolved as <a href="https://logoori.github.io/001.mp3" rel="nofollow noopener">https://logoori.github.io/001.mp3</a>, which when clicked will play the file in the web browser’s media player. You can click the link in the previous sentence to see it work :grinning:</p>
<p>(The reason why the <a href="https://github.com" rel="nofollow noopener">https://github.com</a> link doesn’t work the way you want is because we only enable linking of certain file types and force download of all others, such as MP3s, from <a href="https://github.com" rel="nofollow noopener">https://github.com</a>.)</p>

**Comment by LanguageDescriptions** [comment]
<p>My current question relates to <a href="https://github.community/t/is-it-possible-to-open-a-sound-file/10377/2">this earlier question</a>. My goal is to create a page where users can click on and listen to word recordings (similar to <a href="http://Forvo.com" rel="nofollow noopener">Forvo.com</a>). My earlier strategy was to use an html anchor (&lt;a href=“002.mp3”&gt;play&lt;/a&gt;"), but probably most users (no-tech) will be using phones and this won’t usually or ever work, since browsers will at best offer to download the file. I somewhat sorted out how to use HTML5 and Java to play sounds, and a simple example is at <a href="https://github.com/Logoori/Logoori.github.io/blob/master/4TestJS.htm" rel="nofollow noopener">https://github.com/Logoori/Logoori.github.io/blob/master/4TestJS.htm</a>. It works if I open it locally on my computer.</p>
<p>I tried to open the same thing via <a href="https://htmlpreview.github.io/?https://github.com/Logoori/Logoori.github.io/blob/master/4TestJS.htm" rel="nofollow noopener">https://htmlpreview.github.io/?https://github.com/Logoori/Logoori.github.io/blob/master/4TestJS.htm</a>, and it displays correctly, but it does not play.  The code inside that file is</p>
<pre><code>click &lt;audio id="ID004" source src="004.mp3"&gt;&lt;/audio&gt;&lt;button onclick="playAudio('ID004')" type="button"&gt;play4&lt;/button&gt; and then &lt;audio id="ID003" source src="003.mp3"&gt;&lt;/audio&gt;&lt;button onclick="playAudio('ID003')" type="button"&gt;play3&lt;/button&gt;
&lt;script&gt;
function playAudio(audio_element) {
	var x = document.getElementById(audio_element);
	x.play();
}
&lt;/script&gt;
</code></pre>
<p>I’m trying to figure out why it won’t work. One possible answer is that I’m trying to do something forbidden, and I should stop trying. :mansad: Another thing that I thought of is that I just don’t understand how this html preview thing works, and I’m thinking about this the wrong way. If anyone has any suggestions, that would be highly appreciated.</p>

**Comment by lee-dohm** [comment]
<p>Since this is an extension of the original question, it should just be another post on the same topic so that people can get all the details. I’ve gone ahead and merged the topics for you :grinning:</p>

**Comment by LanguageDescriptions** [comment, marked as answer]
<p>Well, never mind. It turns out that if I just open <a href="https://Logoori.github.io/WHATEVERFILE.HTML" rel="nofollow noopener">https://Logoori.github.io/WHATEVERFILE.HTML</a>, there’s no need for an HTML preview prefix, and the page works as it’s supposed to. <a href="https://github.com/Logoori/Logoori.github.io/blob/master/" rel="nofollow noopener">https://github.com/Logoori/Logoori.github.io/blob/master/</a> seems to have the same stuff as <a href="https://Logoori.github.io" rel="nofollow noopener">https://Logoori.github.io</a>, but it’s treated differently (as a web page).</p>

**Comment by vovs03** [comment]
<h1>404</h1>
<ul>
<li>
<p><code>#404</code></p>
</li>
<li>
<p><code>#NotAvailable</code></p>
</li>
<li>
<p><code>#PageNotFound</code></p>
</li>
</ul>
<p>Please update link to solution</p>

**Comment by LanguageDescriptions** [comment]
<p>I dunno if I can rescind a “solution”. At some later point, the system changed and now it does not work. I suspect the code relied on an unofficial Java quirk. So it doesn’t work anymore, and I’ve abandoned the project for a while.</p>

**Comment by Quacker7** [comment]
<p>Could anyone help me out w/ this? I’m a very amateur coder looking to create my own little soundboard.</p>
<p>I have the “daft” file as an mp3 on my desktop, but can’t get it onto my website. any help is welcomed! Thanks so much!</p>
<p>&lt;body&gt;<br>
&lt;audio controls&gt;<br>
&lt;source src=“daft.mp3” type=“audio/mpeg”&gt;<br>
&lt;/audio&gt;<br>
&lt;/body&gt;</p>

**Comment by noskol-dev** [comment]
<p>I found a way to play audio in github pages</p>
<p>Code:</p>
<pre><code class="lang-auto">&lt;!--This Way Must let Repository Public--&gt;
&lt;script&gt;
var audio = new Audio("https://raw.githubusercontent.com/USERNAME/REPOSITORY/BRANCH/Audio.mp3")
audio.play()
&lt;/script&gt;
</code></pre>

**Reply by Tyler-Hilbert** [reply]
Does this still work? I'm trying this (as of January 2024) and I'm getting a 
_Unhandled Promise Rejection: NotAllowedError: The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission._
error

    <script>
        var audio = new Audio("https://raw.githubusercontent.com/Tyler-Hilbert/3D-Printed-Guitars/main/sound.mp3")
        audio.play()
    </script>
    
    
[https://github.com/Tyler-Hilbert/3D-Printed-Guitars/tree/d63e903165634ed3b9dee4263ef1bcd12bb588ce](    https://github.com/Tyler-Hilbert/3D-Printed-Guitars/tree/d63e903165634ed3b9dee4263ef1bcd12bb588ce)

**Reply by Tyler-Hilbert** [reply]
<script>
      var hasPlayed = false;
      var audio = new Audio(
        "https://raw.githubusercontent.com/Tyler-Hilbert/3D-Printed-Guitars/main/sound.mp3"
      );
      function playAudio() {
        // This func will check if the file hasPlayed and then will disable after first interaction.
        if (!hasPlayed) {
          audio.play();
          hasPlayed = true;
        }
      }
      document.addEventListener("click", playAudio);
    </script>
    
[    @atorrez007 ](https://github.com/atorrez007) came up with this solution
https://github.com/Tyler-Hilbert/3D-Printed-Guitars/commit/23975812508dfb673d4374b19284916157c02f58

**Comment by LanguageDescriptions** [comment]
<p>I sorted out the problem, so here’s a restatement of the problem and the solution. The problem was caused by bad Java. Originally, the page executed the following script:</p>
<p><code>&lt;script&gt;function playAudio(audio_element) {var x = document.getElementById(audio_element); x.play();}&lt;/script&gt;</code></p>
<p>It was then invoked via</p>
<p><code>&lt;audio id="ABC" source src="ABC.wav"&gt;&lt;/audio&gt;&lt;button onclick="playAudio('ABC')" type="button"&gt;play&lt;/button&gt;</code></p>
<p>This worked, then stopped working. The solution (d’oh!) is: take out <em>source</em>.</p>
<p><code>&lt;audio id="ABC" src="ABC.wav"&gt;&lt;/audio&gt;&lt;button onclick="playAudio('ABC')" type="button"&gt;play&lt;/button&gt;</code></p>
<p>On various PCs, the defective code has always worked; so it’s not completely unrelated to GitHub pages, but it was basically a Java problem.</p>

**Comment by jonnyhotchkiss** [comment]
First page opened whilst looking for how to play github wav and other audio stuff from within the browser, vaguely recall seeing a chrome extension to play stuff in / from github, but then skimming this discussion made me think maybe the browser.console can be used to play resources.

Example: https://github.com/kennethreitz/infinite-state-sample-pack/blob/master/Loops/Awaken_116/loop.wav

Then I read further and learned how assets can be played by referencing relative link...
So this mentions a .io site and github pages, but I'm wondering if either 

- the same can be achieved, **without** using pages, eg a github repo with a readme markdown that has links to resources in the repo (.wav, .mp3 etc) that can be clicked to be played (this may already exist)
- (if pages are recommended or required) a workflow that creates the page that accompanies a commit
- (unlimited third-party online storage?) a workflow that uploads a resource to a host like soundcloud (which ever has the best policy, eg closest to free unlimited) and get the link and creates a page that references this

braindump on options for "using sounds in and on github" 📌 

I asked my GPT thing (OpenAI: Ctrl Altman Del 😂)

> GitHub Direct Link:
> Use the raw.githubusercontent.com link to get the direct link to the audio file. In your case, the direct link would be: 

> https://raw.githubusercontent.com/kennethreitz/infinite-state-sample-pack/master/Loops/Awaken_116/loop.wav
> Online Audio Player:
> Use an online audio player that allows you to input a direct link to an audio file. There are various options available, and one popular choice is [OnlineAudioConverter](https://online-audio-converter.com/). Here's how you can use it:

> Visit [OnlineAudioConverter](https://online-audio-converter.com/).
> Click on "Open URL" or a similar option.

So currently a multi-tab setup to drag /drop my work, but even that's more clunk than I'd like... 
I may be missing several obvious things here, a like OP am not a dev or git-jedi (my google-fu is quite strong though:)

**Comment by 212-del** [comment]
This is still not possible yet to display audio like the image in github pages. But we may have the best closest way to do is Use the raw.githubusercontent.com link to get the direct link to the audio file. That means insted of putting the file name in the code [audio file](audio.wav) put the link of audio.wav in the place of the word audio.wav the structure should look like [anything](link of uploaded audio file in your repo.) Till now there is no way to display the audio file in the github pages except the online players links.
